src/frequenz/sdk/actor/_background_service.py

```diff
@@ -0,0 +1,214 @@
+    def name(self) -> str:
+        """The name of this background service.
+
+        Returns:
+            The name of this background service.
+        """
+        return self._name
+
+    @property
+    def tasks(self) -> collections.abc.Set[asyncio.Task[Any]]:
+        """Return the set of running tasks spawned by this background service.
+
+        Users typically should not modify the tasks in the returned set and only use
+        them for informational purposes.
+
+        !!! danger
+
+            Changing the returned tasks may lead to unexpected behavior, don't do it
+            unless the class explicitly documents it is safe to do so.
+
+        Returns:
+            The set of running tasks spawned by this background service.
+        """
+        return self._tasks
+
+    @property
+    def is_running(self) -> bool:
+        """Return whether this background service is running.
+
+        A service is considered running when at least one task is running.
+
+        Returns:
+            Whether this background service is running.
+        """
+        return any(not task.done() for task in self._tasks)
+
+    def cancel(self) -> None:
+        """Cancel all running tasks spawned by this background service."""
+        for task in self._tasks:
+            task.cancel()
+
+    async def stop(self) -> None:
+        """Stop this background service.
+
+        This method cancels all running tasks spawned by this service and waits for them
+        to finish.
+
+        Raises:
+            BaseExceptionGroup: If any of the tasks spawned by this service raised an
+                exception.
+        """
+        self.cancel()
+        done, pending = await asyncio.wait(self._tasks)
+        assert not pending
+        exceptions: list[BaseException] = []
+        for task in done:
+            if task.cancelled():
+                continue
+            if exception := task.exception():
+                exceptions.append(exception)
+        if exceptions:
+            raise BaseExceptionGroup(
+                f"Error while stopping background service {self}", exceptions
+            )
+
+    async def __aenter__(self) -> Self:
+        """Enter an async context.
+
+        Start this background service.
+
+        Returns:
+            This background service.
+        """
+        await self.start()
+        return self
+
+    async def __aexit__(
+        self,
+        exc_type: type[BaseException] | None,
+        exc_val: BaseException | None,
+        exc_tb: TracebackType | None,
+    ) -> None:
+        """Exit an async context.
+
+        Stop this background service.
+
+        Args:
+            exc_type: The type of the exception raised, if any.
+            exc_val: The exception raised, if any.
+            exc_tb: The traceback of the exception raised, if any.
+        """
+        await self.stop()
+
+    def __await__(self) -> collections.abc.Generator[None, None, None]:
+        """Await this background service.
```

Comment:
as in, all tasks that still running? Maybe add this info to the docu

Reply:
Not sure what's unclear, this implements `await backgroud_service`. I just added a clarification that will wait until all the tasks are finished, hope that makes it more clear.